src/main/java/org/spoken_tutorial/health/elasticsearch/JsonService/JsonService.java

```diff
@@ -83,4 +84,165 @@ public String saveNarrationToFile(String url, String documentId) throws ParseExc
+                if (lanId == 22) {
+                    sb.append("<tr>\n<th>\n");
+                    sb.append("Visual Cue");
+                    sb.append("\n</th>\n<th>\n");
+                    sb.append("Narration");
+                    sb.append("\n</th>\n</tr>");
+
+                } else {
+                    sb.append("<tr>\n<th>\n");
+                    sb.append("Time");
+                    sb.append("\n</th>\n<th>\n");
+                    sb.append("Narration");
+                    sb.append("\n</th>\n</tr>");
+                }
+
+                for (int i = 0; i < jsonArrayNarrations.length(); i++) {
+                    JSONObject jsonNarration = (JSONObject) jsonArrayNarrations.get(i);
+                    sb.append("\n<tr>");
+
+                    sb.append((String) jsonNarration.get("cue"));
+
+                    sb.append((String) jsonNarration.get("narration"));
+                    sb.append("\n</tr>");
+
+                }
+                sb.append("\n</table>\n");
+                sb.append("<br>\n");
+                sb.append("</body>\n</html>");
+                String narration = sb.toString();
+
+                Files.writeString(htmlFilePath, narration);
+
+                String temp = htmlFilePath.toString();
+
+                int indexToStart = temp.indexOf("Media");
+
+                document = temp.substring(indexToStart, temp.length());
+
+            }
+
+        } catch (Exception e) {
+            logger.error("Exception Error", e);
+        }
+        return document;
+    }
+
+    public String convertHtmltoOdt(String htmlFie, int tutorialId) {
+
+        String document = "";
+        boolean flag = true;
+
+        try {
+
+            Path odtDir = Paths.get(mediaRoot, Config.uploadDirectoryScriptOdtFileforDownload);
+            Files.createDirectories(odtDir);
+            Path odtfilePath = Paths.get(mediaRoot, Config.uploadDirectoryScriptOdtFileforDownload,
+                    tutorialId + ".odt");
+            File odtfile = odtfilePath.toFile();
+
+            Path htmlDir = Paths.get(mediaRoot, Config.uploadDirectoryScriptHtmlFileforDownload);
+            Files.createDirectories(htmlDir);
+            Path htmlFilePath = Paths.get(mediaRoot, Config.uploadDirectoryScriptHtmlFileforDownload, htmlFie);
+
+            File htmlfile = htmlFilePath.toFile();
+            if (htmlfile.exists() && odtfile.exists()) {
+                if (Files.getLastModifiedTime(odtfilePath).toMillis() > Files.getLastModifiedTime(htmlFilePath)
+                        .toMillis()) {
+                    flag = false;
+                }
+            }
+
+            if (flag) {
+                Path htmlfilePath = Paths.get(mediaRoot, htmlFie);
+                String htmlInputPath = htmlfilePath.toString();
+
+                String odtdirstr = odtDir.toString();
+
+                ProcessBuilder processBuilder = new ProcessBuilder("soffice", "--headless", "--convert-to", "odt",
```

Comment:
does libreoffice also work here instead of soffice? if yes, use it.

Reply:
libreoffice does not work on my system.

Reply:
okay. what error do you get?

add the soffice in property file then. not sure what works on linux.